application/common/models/Reset.php

```diff
@@ -337,4 +377,156 @@ public static function getExpireTimestamp()
 
         return $time + $params['reset']['lifetimeSeconds'];
     }
+
+    /**
+     * Check if this reset is currently disabled
+     * @return bool
+     */
+    public function isDisabled()
+    {
+        if ($this->disable_until !== null) {
+            $disableUntilTime = strtotime($this->disable_until);
+            // Intentionally loose comparison to catch zero
+            if ($disableUntilTime == false) {
+                return true;
+            }
+            return $disableUntilTime > time();
```

Comment:
One thing this logic assumes is that the timezone used for storing `disable_until` in the database is the same timezone as used by `time()`. That may be a safe assumption... but I'm not sure of that. Is there some explicit way we could be using UTC for all of these time values?